filter/specItemFilter.go

```diff
@@ -77,25 +76,15 @@ func (filter *ScenarioFilterBasedOnTags) replaceSpecialChar() {
 }
 
 func (filter *ScenarioFilterBasedOnTags) formatAndEvaluateExpression(tagsMap map[string]bool, isTagQualified func(tagsMap map[string]bool, tagName string) bool) (bool, error) {
-	_, tags := filter.getOperatorsAndOperands()
-	expToBeEvaluated := filter.tagExpression
-	sort.Sort(ByLength(tags))
+	tagExpressionParts, tags := filter.parseTagExpression()
```

Comment:
Why are these changes showing up in this PR. These changes are already merged in #671